packages/flutter/test/material/carousel_test.dart

```diff
@@ -1149,6 +1149,48 @@ void main() {
     expect(getItem(5), findsOneWidget);
     expect(tester.getRect(getItem(5)).width, difference);
   });
+
+  testWidgets('Updating Carousel does not cause exception', (WidgetTester tester) async {
+    bool isLight = true;
+    await tester.pumpWidget(
+      StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return MaterialApp(
+            theme: Theme.of(context).copyWith(
+              brightness: isLight ? Brightness.light : Brightness.dark,
+            ),
+            home: Scaffold(
+              appBar: AppBar(
+                actions: <Widget>[
+                  Switch(
+                    value: isLight,
+                    onChanged: (bool value) {
+                      setState(() {
+                        isLight = value;
+                      });
+                    }
+                  )
+                ],
+              ),
+              body: CarouselView(
+                itemExtent: 100,
+                children: List<Widget>.generate(10, (int index) {
+                  return Center(
+                    child: Text('Item $index'),
+                  );
+                }),
+              ),
+            ),
+          );
+        }
+      )
+    );
+    await tester.pumpAndSettle();
+    await tester.tap(find.byType(Switch));
+    await tester.pumpAndSettle();
+
+    // No exception.
```

Comment:
You can use an except to verify this :)

```suggestion
    expect(tester.takeException(), isNull);
```